resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/HttpClientUtil.scala

```diff
@@ -33,6 +33,7 @@ private[spark] object HttpClientUtil {
 
   def createClient[T: ClassTag](
       uris: Array[String],
+      maxRetriesPerServer: Int = 1,
       sslSocketFactory: SSLSocketFactory = SSLContext.getDefault.getSocketFactory,
       trustContext: X509TrustManager = null,
       readTimeoutMillis: Int = 20000,
```

Comment:
I saw a cute trick recently for status checking we could use below.  Instead of:

`if (response.status() >= 200 && response.status() < 300) {`

use

`if (response.status() / 100 == 2) {`